src/SlimBootstrap/Authentication/Jwt.php

```diff
@@ -52,8 +58,14 @@ public function __construct(
         $this->clientDataClaims = $clientDataClaims;
         $this->claimsConfig     = $claimsConfig;
         $this->logger           = $logger;
-    }
 
+        $this->jwtConfig = Lcobucci\JWT\Configuration::forAsymmetricSigner(
+            $this->determineSigner($this->encryption),
+            Lcobucci\JWT\Signer\Key\InMemory::plainText(''), // setting signKey empty, as we only verify tokens here
+            Lcobucci\JWT\Signer\Key\InMemory::plainText($publicKey)
+        );
+        $this->jwtConfig->setValidationConstraints(...$this->evaluateJwtContstrains());
```

Comment:
Why do we need to set the validation constraints in the constructor? can't we do this in the authentication class before they are used? I would like to avoid too much logic in the constructor

Reply:
I think the constructor is fitting, as it prepares and initializes the `Lcobucci\JWT\Configuration` object for its later use. In my opinion, putting the logic into the `authenticate` is easily possible, but I kind of expect, that all needed objects are already ready to use at this point. Alternatively, I think this logic could also be moved into the `Authentication/Factory->createJwt()` to avoid having this logic in the constructor.

Reply:
> I think this logic could also be moved into the Authentication/Factory->createJwt() to avoid having this logic in the constructor.

And then inject the `jwtConfig` object into the `Authentication\Jwt` class? Hm maybe.... also what kind of errors can occur in these new calls? Any possible exceptions we should be aware about?

Reply:
> And then inject the jwtConfig object into the Authentication\Jwt class?

correct 

>  also what kind of errors can occur in these new calls? Any possible exceptions we should be aware about?

I did not saw, that exception will be thrown in these method calls. Also the example in the documentation does not show any hints, that these calls should be in a try/catch block.

Reply:
I have moved the logic now in the authentication method, as otherwise, the Sombra authentication would break. I think that this authentication method is not used anymore, but I only want to release a bugfix version here.